components/screens/BlogScreen/BlogScreen.js

```diff
@@ -1,12 +1,80 @@
+import { useIsFocused } from '@react-navigation/native';
 import React from 'react';
-import { View, Text } from 'react-native';
+import {useState, useEffect} from 'react';
+import {View, Text, ScrollView, Linking, Button} from 'react-native';
+import { Image } from 'react-native-elements';
+import InAppBrowser from 'react-native-inappbrowser-reborn';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import {Styles} from './BlogStyleSheet';
+
+var blogData = [];
+
+// async function getBlogData() {
+//   let response = await fetch('https://cascade-api.herokuapp.com/blog');
+//   let dataPromise = await response.json();
+//   dataPromise.forEach(data => blogData.push(data));
+
+//   // console.log('OUTSIDE', blogData);
+// }
+// getBlogData();
+
 
 const BlogScreen = () => {
+  
+    const [data, setData] = useState([]);
+    const isFocused = useIsFocused();
+  
+    const getData = async () => {
+      const resp = await fetch("https://cascade-api.herokuapp.com/blog");
+      const data = await resp.json();
+      setData(data);
+      console.log(data);
+    };
+    useEffect(() => {
+      console.log('useEffect called');
+      if(isFocused) {
+      getData();
+     
+      }
+
+    }, [isFocused]);
+ 
   return (
-    <View>
-      <Text>Blog View</Text>
-    </View>
+    <SafeAreaView>
+    <ScrollView>
+     
+      {data.map(({title, postDate, body, url}) => (
+         <View key = {url}>
+          <Text key={title} style={Styles.title}>
+            {title}
+          </Text>
+          <Text style={Styles.date}
+          key={postDate}>{postDate}</Text>
+          <Image key={'rc34'}
+          source={{uri: 'https://cbc-drupal-assets.s3.amazonaws.com/styles/blog_image/s3/blog/leafy%20and%20dank%20%283%29.JPG?R2Pc9TRNWsiPfO0asl_VSPULtFDI4Dqs&itok=O8FLO9W1'}}
+          style={Styles.image}
+          
+           />
+          <Text key={body} style={Styles.blogItem} >{body}</Text>
+          <View key={url} style = {{ width: 200, height: 50 }}>
+          <Button
+          style={Styles.button}
+          type="solid"
+          title="Read more"
+          color="#F55243"
+          size={6}
+    onPress={(e) => {
+      e.preventDefault();
+      InAppBrowser.open('http://www.cascade.org' + url);
+      }}
+/>
+</View>
+        </View>
+      ))
```

Comment:
**Nitpick**: Could we clean up the formatting a little in this file?

VSCode has auto-formatting for most file types. You can also get extensions for other file types that are not naturally supported. For me, the auto-formatting is `Alt` + `Shift` + `F`. _(This could be different for you.)_

Check [here](https://newbedev.com/how-do-you-format-code-in-visual-studio-code-vscode) if you can't find it.